olive/engine/engine.py

```diff
@@ -814,31 +815,34 @@ def _load_model(self, model_id: str) -> Union[OliveModel, str]:
         if model_json == {}:
             return FAILED_CONFIG
 
-        model = ModelConfig.from_json(model_json).create_model()
+        model = ModelConfig.from_json(model_json)
         return model
 
-    def _prepare_non_local_model(self, model: OliveModel) -> OliveModel:
+    def _prepare_non_local_model(self, model_config: ModelConfig) -> ModelConfig:
         """
         Prepare models with non-local model path for local run by downloading the model resources to cache
         """
-        for resource_name, resource_path in model.resource_paths.items():
+        resource_keys = model_config.get_resource_keys()
+        for resource_name in resource_keys:
+            resource_path = model_config.config.get(resource_name)
+            resource_path = create_resource_path(resource_path)
```

Comment:
This can be simplfied using `model_config.get_resource_paths` like in other places in this PR.

Reply:
will change it